src/engine/threaded_engine.cc

```diff
@@ -415,6 +415,25 @@ void ThreadedEngine::WaitForAll() {
   finished_cv_.wait(lock, [this]() {
       return pending_.load() == 0 || kill_.load();
     });
+  std::exception_ptr tmp;
+  if (!global_exception_refs_.empty()) {
+    // iterate through all exception refs
+    for (auto itr = global_exception_refs_.begin();
+         itr != global_exception_refs_.end(); ++itr) {
+      const ExceptionRef& ptr = *itr;
+      // the first exception will be saved to be rethrown later
+      if (*ptr != nullptr && !tmp) {
```

Comment:
can be evaluated in bool context, so less noise.

Reply:
changed